anonymizer/OpenDiffix.Service.Tests/Program.Tests.fs

```diff
@@ -5,79 +5,62 @@ open FsUnit.Xunit
 
 open OpenDiffix.Service.Program
 
-let private dataDirectory = __SOURCE_DIRECTORY__ + "/../../data/data.sqlite"
+let private dataPath = __SOURCE_DIRECTORY__ + "/../reference/data/customers.csv"
 
-// [<Fact>]
-// let ``Prints version`` () =
-//   [| "--version" |] |> mainCore |> should not' (be Empty)
+type TempFile() =
+  let path = System.IO.Path.GetTempFileName()
+  member this.Path = path
 
-// [<Fact>]
-// let ``Counts all rows`` () =
-//   [| "-f"; dataDirectory; "--aid-columns"; "customers.id"; "-q"; "SELECT count(*) FROM customers" |]
-//   |> mainCore
-//   |> should not' (be Empty)
+  interface System.IDisposable with
+    member this.Dispose() = System.IO.File.Delete(path)
 
-// [<Fact>]
-// let ``Counts in non-anonymized tables`` () =
-//   [| "-f"; dataDirectory; "-q"; "SELECT count(*) FROM purchases" |]
-//   |> mainCore
-//   |> should not' (be Empty)
+[<Fact>]
+let ``Handles Load request`` () =
+  let request =
+    $"""
+    {{"type":"Load","inputPath":"%s{dataPath}","rows":10000}}
+    """
 
-// [<Fact>]
-// let ``Rejects invalid SQL`` () =
-//   shouldFail (fun () ->
-//     [|
-//       "-f"
-//       dataDirectory
-//       "--aid-columns"
-//       "customers.id"
-//       "-q"
-//       "SELECT no_such_column, count(*) FROM customers GROUP BY no_such_column"
-//     |]
-//     |> mainCore
-//     |> ignore
-//   )
+  let response = request |> mainCore |> Option.get
+  response |> should haveSubstring "columns"
+  response |> should haveSubstring "rows"
 
-// [<Fact>]
-// let ``Guards against unknown params`` () =
-//   shouldFail (fun () ->
-//     [| "-f"; dataDirectory; "--foo"; "customers.id"; "-q"; "SELECT count(*) FROM customers" |]
-//     |> mainCore
-//     |> ignore
-//   )
+[<Fact>]
+let ``Handles HasMissingValues request`` () =
+  let request =
+    $"""
+    {{"type":"HasMissingValues","inputPath":"%s{dataPath}","aidColumn":"\"id\""}}
+    """
 
-// [<Fact>]
-// let ``Accepts supported CLI parameters`` () =
-//   [|
-//     "-f"
-//     dataDirectory
-//     "--salt"
-//     "1"
-//     "--json"
-//     "--outlier-count"
-//     "1"
-//     "2"
-//     "--top-count"
-//     "12"
-//     "14"
-//     "--low-threshold"
-//     "3"
-//     "--low-sd"
-//     "1.2"
-//     "--low-mean-gap"
-//     "1"
-//     "--noise-sd"
-//     "2.4"
-//     "--aid-columns"
-//     "customers.id"
-//     "-q"
-//     "SELECT count(*) FROM customers"
-//   |]
-//   |> mainCore
-//   |> should not' (be Empty)
+  [ "true"; "false" ] |> should contain (request |> mainCore |> Option.get)
 
-// [<Fact>]
-// let ``Executes example batch query`` () =
-//   [| "--queries-path"; __SOURCE_DIRECTORY__ + "/../../queries-sample.json" |]
-//   |> mainCore
-//   |> should not' (be Empty)
+[<Fact>]
+let ``Handles Preview request`` () =
+  let request =
+    $"""
+    {{"type":"Preview","inputPath":"%s{dataPath}","aidColumn":"\"id\"","salt":"1","buckets":["\"age\"","\"city\""],"countInput":"Rows","rows":1000}}
```

Comment:
It would be a bit more readable to instead create the input as a map and then jsonify it.

Reply:
I considered this but rejected the idea b/c:
1/ it's now super-easy to just copy paste these requests from manual testing logs
2/ JSON seemed to require significant boilerplate, judging by the `JSONDecodersEncoders.fs` work I did recently, but maybe if I use `Map` here (which is what you suggest, right), it's not that bad?

I'll give it another thought

Reply:
hm, to me both alternatives are worse off

```
  let request =
    $"""
    {{"type":"HasMissingValues","inputPath":"%s{dataPath}","aidColumn":"\"id\""}}
    """

  // vs
  let requestMap = Map [ ("type", "HasMissingValues"); ("inputPath", dataPath); ("aidColumn", "id") ]
  let request = Encode.Auto.toString (2, requestMap)

  // vs
  let requestMap = {| Type = "HasMissingValues"; InputPath = dataPath; AidColumn = "id"|}
  let request = Encode.Auto.toString (2, requestMap, caseStrategy = CamelCase)
```

Reply:
oh, and I noticed that the escaping is redundant in the string version 7b1bc2b7

Reply:
Personally, I like the anon type version more. But I won't insist on it.